sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/AggregationQuerySuite.scala

```diff
@@ -545,19 +576,21 @@ abstract class AggregationQuerySuite extends QueryTest with SQLTestUtils with Te
           |  count(distinct value2),
           |  sum(distinct value2),
           |  count(distinct value1, value2),
+          |  longProductSum(distinct value1, value2),
```

Comment:
Yes and No.

The input we care about only consists of these tuples: ```[value1=null, value2=null], [value1=null, value2=1], [value1=1, value2=null], and [value1=1, value2=1]```

However in the current implementation a distinct aggregate will see more input than those. It will also see records from other groups. However, the values in these records are nulled out. The assumption here is that an AggregateFunction is not changed by an all NULL update. The only case I can think of that would be problematic is a ```FIRST(DISTINCT ...)```; which shouldn't be used like that anyway.

We could solve this by wrapping AggregateFunctions with an operator which will only update if the group id is correct.